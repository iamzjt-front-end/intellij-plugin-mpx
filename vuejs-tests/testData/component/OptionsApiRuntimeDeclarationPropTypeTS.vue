<script lang="ts">
import { PropType } from 'vue'

interface Book {
  title: string
  author: string
  year: number
}

export default {
  name: "TestComponent",
  props: {
    book: { type: Object as PropType<Book>, required: true }
  }
}
</script>

<template>
  <TestComponent :book="<weak_warning descr="Type {year: number, title: string} is not assignable to type Book">{title: 'WebStorm', year: 2022}</weak_warning>" />
  <TestComponent :book="{title: 'WebStorm', year: 2022, author: 'JB'}" />
</template>
