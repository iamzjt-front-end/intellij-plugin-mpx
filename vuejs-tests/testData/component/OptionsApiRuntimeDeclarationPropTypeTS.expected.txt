{
 class: "VueSourceComponent",
 defaultName: "TestComponent",
 props: [
  {
   class: "VueSourceInputProperty",
   name: "book",
   jsType: "Book",
   required: true,
  },
 ],
}
